src/main/java/org/opentripplanner/middleware/controllers/api/ApiUserController.java

```diff
@@ -99,25 +100,30 @@ private ApiUser createApiKeyForApiUser(Request req, Response res) {
             }
         }
         // FIXME Should an Api user be limited to one api key per usage plan (and perhaps stage)?
-        ApiKey apiKey = ApiGatewayUtils.createApiKey(targetUser.id, usagePlanId);
-        if (apiKey == null || apiKey.keyId == null) {
+        try {
+            ApiKey apiKey = ApiGatewayUtils.createApiKey(targetUser, usagePlanId);
+            // Add new API key to user and persist
+            targetUser.apiKeys.add(apiKey);
+            Persistence.apiUsers.replace(targetUser.id, targetUser);
+        } catch (CreateApiKeyException e) {
             logMessageAndHalt(req,
                 HttpStatus.INTERNAL_SERVER_ERROR_500,
-                String.format("Unable to get AWS API key for user id (%s) and usage plan id (%s)", targetUser.id, usagePlanId),
-                null
+                "Error creating API key",
+                e
             );
-            return null;
         }
-        // Add new API key to user and persist
-        targetUser.apiKeys.add(apiKey);
-        Persistence.apiUsers.replace(targetUser.id, targetUser);
         return Persistence.apiUsers.getById(targetUser.id);
     }
 
     /**
      * Delete an api key from a given user's list of api keys (if present) and from AWS api gateway.
      */
     private ApiUser deleteApiKeyForApiUser(Request req, Response res) {
+        Auth0UserProfile requestingUser = Auth0Connection.getUserFromRequest(req);
+        // Do not permit key deletion unless user is an admin.
+        if (!isUserAdmin(requestingUser)) {
```

Comment:
As a result of this new restriction, test `ApiKeyManagementTest#canDeleteApiKeyForSelf()` should be removed.

Reply:
Good catch. How about we change it to cannotDeleteApiKeyForSelf?

Reply:
That should do too, and we'll have a test to police the new restriction.